src/utilities/schemaComparisons.js

```diff
@@ -0,0 +1,220 @@
+    const newType: GraphQLType = newTypeMap[typeName];
+    if (!(oldType instanceof newType.constructor)) {
+      typesThatChangedType.push(`${typeName} changed from a ` +
+        `${oldType.constructor.name} to a ${newType.constructor.name}`);
+    }
+  }
+  return new Set(typesThatChangedType);
+}
+
+/**
+* Given two schemas, returns a Set containing descriptions of any breaking
+* changes in the newSchema related to the fields on a type. This includes if
+* a field has been removed from a type or if a field has changed type.
+*/
+export function findBreakingFieldChanges(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema
+): Set<string> {
+  const oldTypeMap: TypeMap = oldSchema.getTypeMap();
+  const newTypeMap: TypeMap = newSchema.getTypeMap();
+
+  const breakingFieldChanges: Array<string> = [];
+  for (const typeName: string in oldTypeMap) {
+    if (!{}.hasOwnProperty.call(oldTypeMap,typeName)) {
+      continue;
+    }
+    const oldType:GraphQLType = oldTypeMap[typeName];
+    const newType:GraphQLType = newTypeMap[typeName];
+    if (
+      !(oldType instanceof GraphQLObjectType ||
+        oldType instanceof GraphQLInterfaceType ||
+        oldType instanceof GraphQLInputObjectType) ||
+      !(newType instanceof oldType.constructor)
+    ) {
+      continue;
+    }
+
+    const oldTypeFieldsDef: InputObjectFieldMap | GraphQLFieldDefinitionMap =
+      oldType.getFields();
+    const newTypeFieldsDef: InputObjectFieldMap | GraphQLFieldDefinitionMap =
+      newType.getFields();
+    for (const fieldName: string in oldTypeFieldsDef) {
+      // Check if the field is missing on the type in the new schema.
+      if (!(fieldName in newTypeFieldsDef)) {
+        breakingFieldChanges.push(`${typeName}.${fieldName} was removed`);
+      } else {
+        // Check if the field's type has changed in the new schema.
+        const oldFieldType: ?GraphQLNamedType =
+          getNamedType(oldTypeFieldsDef[fieldName].type);
+        const newFieldType: ?GraphQLNamedType =
+          getNamedType(newTypeFieldsDef[fieldName].type);
+        if (
+          oldFieldType && newFieldType &&
+          (oldFieldType.name !== newFieldType.name)
+        ) {
+          breakingFieldChanges.push(`${typeName}.${fieldName} changed type ` +
+            `from ${oldFieldType.name} to ${newFieldType.name}`);
+        }
+      }
+    }
+  }
+  return new Set(breakingFieldChanges);
+}
+
+/**
+* Given two schemas, returns a Set containing descriptions of any breaking
+* changes in the newSchema related to removing types from a union type.
+*/
+export function findTypesRemovedFromUnions(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema
+): Set<string> {
+  const oldTypeMap: TypeMap = oldSchema.getTypeMap();
+  const newTypeMap: TypeMap = newSchema.getTypeMap();
+
+  const typesRemovedFromUnion: Array<string> = [];
+  for (const typeName: string in oldTypeMap) {
+    if (!{}.hasOwnProperty.call(oldTypeMap,typeName)) {
```

Comment:
I had that, but then I got `TypeError: oldTypeMap.hasOwnProperty is not a function`. I guess Flow-typed objects don't inherit `hasOwnProperty`?

Doesn't really matter though since I took your advice above and converted to forEach loops.